lisa/nic.py

```diff
@@ -284,6 +310,15 @@ def reload(self) -> None:
         self.nics.clear()
         self._initialize()
 
+    def check_sriov_enabled(self, sriov_enabled: bool) -> None:
+        if sriov_enabled and not self.get_lower_nics():
+            raise LisaException("SRIOV is enabled, but VF is not found.")
+        elif not sriov_enabled and self.get_lower_nics():
+            raise LisaException("SRIOV is disabled, but VF exists.")
+        else:
```

Comment:
Refine the logic as below, it's clearer to understand they are independenc logic.

```python
assert_that(sriov_enabled).described_as("AN is enabled, but pci device is not found.").not_equals(self.get_lower_nics())
assert_that(sriov_enabled).described_as("AN is not enabled, but pci device is found.").equals(self.get_lower_nics())
```